fn.go

```diff
@@ -0,0 +1,230 @@
+package secp256k1
+
+/*
+
+// NOTE: Requires the gmp c library.
+#define USE_NUM_GMP
+
+#define HAVE___INT128
+#define USE_SCALAR_4X64
+#define USE_SCALAR_INV_BUILTIN
+
+#include "secp256k1/include/secp256k1.h"
+#include "secp256k1/src/util.h"
+#include "secp256k1/src/num_gmp_impl.h"
+#include "secp256k1/src/scalar.h"
+#include "secp256k1/src/scalar_impl.h"
+#include "secp256k1/src/scalar_4x64_impl.h"
+
+*/
+import "C"
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+	"math/big"
+	"unsafe"
+
+	"github.com/renproject/surge"
+)
+
+// Fn represents an element of the field defined by the prime N, where N is the
+// order of the elliptic curve group secp256k1.
+type Fn struct {
+	inner C.secp256k1_scalar
+}
+
+// NewFnFromUint returns a new field element equal to the given unsigned
+// integer.
+func NewFnFromUint(v uint16) Fn {
```

Comment:
Consider renaming to `NewFnFromUint16` or `NewFnFromU16` to be more precise about the input type. Later, we may find ourselves wanting to be able to construct `Fn` from `uint32` or `uint64` and a name like `NewFnFromU16` is more forwards-compatible.